doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+ * operator-sdk scorecard -l ‘testgroup=required’ --fail-fast
+
+Tasks for this story include:
+ * change scorecard by removing earnedPoints and maximumPoints from each test
+ * change scorecard output by removing totalScorePercent, partialPass, earnedPoints, maximumPoints from scorecard output
+ * change scorecard existing tests to remove partialPass, instead only returning a pass or fail
+ * change scorecard CLI to return an exit code based on the pass/fail results from the set of executed tests 
+ * (stretch goal) add support for --fast-fail CLI flag
+
+
+Rough point estimate is 8 points.
+
+
+#### Story 2 - Change Scorecard Test Selection
+Today, the scorecard lets you select which tests are run by the plugins defined in the scorecard configuration (basic, olm). This story would change how users determine which tests are executed by scorecard.
+
+This story would introduce labels for each test and then allow the scorecard CLI to filter which tests it runs based on label selectors.
+
+Tests can fall into 3 groups: required, recommended, or optional. Tests can also be categorized as static or runtime. Labels for these groups and categories would allow a test to be more precisely specified by an end user.
+
+Possible examples of specifying what tests to run are as follows: 
+ * operator-sdk scorecard -l ‘testgroup in (required,recommended,optional)’ 
+ * operator-sdk scorecard -l ‘testtype in (runtime,static)’
+ * operator-sdk scorecard -l updatecrtest=true
+ * operator-sdk scorecard -l someplugintest=true
+
+Tasks for this story include:
+ * Update the test interfaces to support arbitrary test-defined labels
+ * Add the labels to the tests
+ * Update the CLI and scorecard library to support label selectors to filter scorecard runs
+ * A stretch goal would include the ability of the CLI to perform a dry-run of tests based on label selectors to show the user which tests would be executed for a given label selector value.
+
+Rough point estimate for this story is 8 points.
+#### Story 3 - Common Validation
+The scorecard would use a common validation codebase to verify bundle contents. This story supports a single “source of truth” and avoids duplication or variance of validation logic. The scorecard user would be able to specify a bundle on the command line, that bundle would then be validated.  For example:
+ * operator-sdk scorecard --bundle ./my-operator
+
+
+Tasks for this story include:
+ * determine the correct validation library to use
+ * add support for a bundle directory flag to the CLI
+ * change the scorecard code to use that validation library when validating a CR, CSV, or given a bundle directory
+ * format the validation output to present to the CLI user in text or json
+
+Rough point estimate in this story is 5 points.
+
+#### Story 4 - Update scorecard Documentation
+Tasks:
+ * Document changes to CLI
```

Comment:
IMHO We should not have the cli commands documented in the docs instead of it we should have a --help command and some automatically doc as it was done [here](https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/Makefile#L73)  for example. 

<img width="904" alt="Screenshot 2019-09-18 at 21 23 30" src="https://user-images.githubusercontent.com/7708031/65225299-d1cb9e00-dabc-11e9-83a4-9d543c4ede77.png">

Reply:
I think it should be both, document the changes but we need to make sure the help text is accurate.

Reply:
I believe the CLI library we use supports generating docs from the CLI configuration, and I think it's in our backlog to investigate using that to keep our SDK CLI reference up-to-date. I could see this being something we contribute to upstream and then inherit as part of our Kubebuilder integration plan.